lib/api_catalogue.rb

```diff
@@ -10,6 +11,56 @@ def self.from_csv(catalogue_csv:, organisation_csv:)
     new(apis: apis, organisations: organisations)
   end
 
+  def self.from_url(url)
+    url += "/apis"
+
+    uri = URI(url)
+
+    req = Net::HTTP::Get.new(uri)
+    req["correlation-id"] = SecureRandom.uuid
+    req["accept"] = "application/vnd.uk.gov.api.v1alpha+json"
+
+    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
+      http.request(req)
+    end
+
+    raise "Could not parse APIs from http://localhost/subpath, no version supported" if response.is_a? Net::HTTPNotAcceptable
+
+    response = JSON.parse response.body
+
+    apis = response["apis"].map do |api|
+      data = api["data"]
+      Api.new name: data["name"],
+              description: data["description"],
+              url: data["url"],
+              documentation: data["documentation-url"],
+              provider: data["organisation"],
+              date_added: Date.today,
+              date_updated: Date.today
+    end
+
+    orgs = response["apis"].map do |api|
+      Organisation.new id: api["data"]["organisation"], # TODO
+                       name: api["data"]["organisation"],
+                       alternate_name: api["data"]["organisation"]
+    end
+    new(apis: apis, organisations: orgs)
+  end
+
+  def self.from_csv_and_urls(catalogue_csv, organisation_csv, *urls)
```

Comment:
I would probably have done this differently, instead letting the caller do something like:

```ruby
static = ApiCatalogue.from_csv(catalogue_csv: catalogue_csv, organisation_csv: organisation_csv)
sandbox = ApiCatalogue.from_url('https://federated-api-model-spring-boot-sandbox.london.cloudapps.digital')
dwp = ApiCatalogue.from_url('...')

api_catalogue = ApiCatalogue.merge(static, sandbox, dwp)

# then use a single `api_catalogue`
```

Reply:
(We'd likely want to add something in here for error handling - no point failing the build if one of the external APIs is down)